From a862e751684d215ba10eb1146fb876ae9f281ca5 Mon Sep 17 00:00:00 2001
From: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
Date: Thu, 30 Sep 2021 16:31:19 +0300
Subject: [PATCH 159/322] [FIO internal] imx8qm_mek: move setting bootcmd to
 defconfig

Move setting the default boot command to the imx8qm_mek_defconfig. It
allows replacing the command without code modification.

Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
---
 configs/imx8qm_mek_defconfig |  3 ++-
 include/configs/imx8qm_mek.h | 19 -------------------
 2 files changed, 2 insertions(+), 20 deletions(-)

diff --git a/configs/imx8qm_mek_defconfig b/configs/imx8qm_mek_defconfig
index f8fa8ee0e51..4e03b46300c 100644
--- a/configs/imx8qm_mek_defconfig
+++ b/configs/imx8qm_mek_defconfig
@@ -28,6 +28,7 @@ CONFIG_PANIC_HANG=y
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=board/freescale/imx8qm_mek/imximage.cfg"
 CONFIG_BOOTDELAY=3
+CONFIG_BOOTCOMMAND="mmc dev ${mmcdev}; if mmc rescan; then if run loadbootscript; then run bootscript; else if test ${sec_boot} = yes; then if run loadcntr; then run mmcboot; else run netboot; fi; else if run loadimage; then run mmcboot; else run netboot; fi; fi; fi; else booti ${loadaddr} - ${fdt_addr}; fi"
 CONFIG_LOG=y
 CONFIG_BOARD_EARLY_INIT_F=y
 CONFIG_SPL_BOARD_INIT=y
@@ -184,4 +185,4 @@ CONFIG_VIDEO_IT6263_BRIDGE=y
 CONFIG_SYS_WHITE_ON_BLACK=y
 CONFIG_SPLASH_SCREEN=y
 CONFIG_SPLASH_SCREEN_ALIGN=y
-CONFIG_CMD_BMP=y
\ No newline at end of file
+CONFIG_CMD_BMP=y
diff --git a/include/configs/imx8qm_mek.h b/include/configs/imx8qm_mek.h
index e90b2508dbd..867d445ace4 100644
--- a/include/configs/imx8qm_mek.h
+++ b/include/configs/imx8qm_mek.h
@@ -257,25 +257,6 @@
 			"fi;" \
 		"fi;\0"
 
-#define CONFIG_BOOTCOMMAND \
-	   "mmc dev ${mmcdev}; if mmc rescan; then " \
-		   "if run loadbootscript; then " \
-			   "run bootscript; " \
-		   "else " \
-			   "if test ${sec_boot} = yes; then " \
-				   "if run loadcntr; then " \
-					   "run mmcboot; " \
-				   "else run netboot; " \
-				   "fi; " \
-			    "else " \
-				   "if run loadimage; then " \
-					   "run mmcboot; " \
-				   "else run netboot; " \
-				   "fi; " \
-			 "fi; " \
-		   "fi; " \
-	   "else booti ${loadaddr} - ${fdt_addr}; fi"
-
 /* Link Definitions */
 #ifdef CONFIG_TARGET_IMX8QM_MEK_A72_ONLY
 	#define CONFIG_LOADADDR			0xC0280000
-- 
2.34.1

